src/lib.rs

```diff
@@ -149,33 +175,90 @@ pub enum Resource {
 }
 
 impl Resource {
+    /// Returns the raw resource type
     #[inline(always)]
-    pub fn as_raw_resource(&self) -> RawResource {
-        *self as _
+    pub const fn as_raw_resource(self) -> RawResource {
+        self as _
     }
 
     /// Set resource limits.
     #[inline(always)]
-    pub fn set(&self, soft: rlim, hard: rlim) -> std::io::Result<()> {
-        setrlimit(*self, soft, hard)
+    pub fn set(self, soft: rlim, hard: rlim) -> io::Result<()> {
+        setrlimit(self, soft, hard)
     }
 
     /// Get resource limits.
     #[inline(always)]
-    pub fn get(&self) -> std::io::Result<(rlim, rlim)> {
-        getrlimit(*self)
+    pub fn get(self) -> std::io::Result<(rlim, rlim)> {
+        getrlimit(self)
+    }
+}
+
+impl std::str::FromStr for Resource {
+    type Err = RlimitsResourceError;
+
+    /// Parse a `Resource` from a &str
+    #[inline(always)]
+    fn from_str(s: &str) -> std::result::Result<Self, Self::Err> {
+        match s {
+            #[cfg(any(target_os = "linux", target_os = "macos", target_os = "ios"))]
+            "RLIMIT_AS" => Ok(Self::AS),
+
+            "RLIMIT_CORE" => Ok(Self::CORE),
+
+            "RLIMIT_CPU" => Ok(Self::CPU),
+
+            "RLIMIT_DATA" => Ok(Self::DATA),
+
+            "RLIMIT_FSIZE" => Ok(Self::FSIZE),
+
+            #[cfg(target_os = "linux")]
+            "RLIMIT_LOCKS" => Ok(Self::LOCKS),
+
+            #[cfg(any(target_os = "linux", target_os = "macos", target_os = "ios"))]
+            "RLIMIT_MEMLOCK" => Ok(Self::MEMLOCK),
+
+            #[cfg(target_os = "linux")]
+            "RLIMIT_MSGQUEUE" => Ok(Self::MSGQUEUE),
+
+            #[cfg(target_os = "linux")]
+            "RLIMIT_NICE" => Ok(Self::NICE),
+
+            #[cfg(any(target_os = "linux", target_os = "macos", target_os = "ios"))]
+            "RLIMIT_NOFILE" => Ok(Self::NOFILE),
+
+            #[cfg(any(target_os = "linux", target_os = "macos", target_os = "ios"))]
+            "RLIMIT_NPROC" => Ok(Self::NPROC),
+
+            #[cfg(target_os = "linux")]
+            "RLIMIT_RSS" => Ok(Self::RSS),
+
+            #[cfg(target_os = "linux")]
+            "RLIMIT_RTPRIO" => Ok(Self::RTPRIO),
+
+            #[cfg(all(target_os = "linux", target_env = "gnu"))]
+            "RLIMIT_RTTIME" => Ok(Self::RTTIME),
+
+            #[cfg(target_os = "linux")]
+            "RLIMIT_SIGPENDING" => Ok(Self::SIGPENDING),
+
+            "RLIMIT_STACK" => Ok(Self::STACK),
```

Comment:
Do we need the `RLIMIT_` prefix?

Reply:
Yes, we need the prefix because users may pass `RLIMIT_` flags to the `from_str` like arguments of `setrlimit` system call.
If we consider reading the rlimit flags from a configuration file, the flags in that file are likely to have `RLIMIT_` prefix like linux api.

Reply:
Ok